pennylane/ftqc/lattice.py

```diff
@@ -0,0 +1,189 @@
+        """
+        # TODO: This method could be renamed later as it's possible that this method is only for the entanglement setup.
+        return nx.get_edge_attributes(self._graph, attribute_name)
+
+    def get_neighbors(self, node):
+        r"""Returns the neighbors of a given node in the lattice.
+
+        Args:
+            node: a target node label.
+        """
+        return self._graph.neighbors(node)
+
+    def get_nodes(self):
+        r"""Returns all nodes in the lattice."""
+        return self._graph.nodes
+
+    def get_edges(self):
+        r"""Returns all edges in the lattice."""
+        return self._graph.edges
+
+    def get_graph(self) -> nx.Graph:
+        r"""Returns the underlying NetworkX graph object representing the lattice."""
+        return self._graph
+
+
+def generate_lattice(lattice, dims: List[int]) -> Lattice:
+    r"""Generates a :class:`~pennylane.ftqc.Lattice` object for a given lattice shape and dimensions.
+
+    Args:
+        lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'rectangle'``, ``'honeycomb'``, ``'triangle'``, ``'cubic'``.
+        dims(List[int]): Number of nodes in each direction of the lattice.
+
+    Returns:
+        a :class: `~pennylane.ftqc.Lattice` object.
+
+    Raises:
+        ValueError: If the lattice shape is not supported or the dimensions are invalid.
+    """
+    # TODOs: Add default support to CSS, Shor, Foliation code lattices.
+
+    lattice_shape = lattice.strip().lower()
+
+    if lattice_shape not in ["chain", "rectangle", "honeycomb", "triangle", "cubic"]:
+        raise ValueError(
+            f"Lattice shape, '{lattice}' is not supported."
+            f"Please set lattice to: 'chain', 'rectangle', 'honeycomb', 'triangle', 'cubic'."
+        )
+
+    if lattice_shape in ["chain", "rectangle", "cubic"]:
+        if lattice_shape == "chain" and len(dims) != 1:
+            raise ValueError(
+                f"For a chain lattice, the length of dims should 1 instead of {len(dims)}"
+            )
+
+        if lattice_shape == "rectangle" and len(dims) != 2:
+            raise ValueError(
+                f"For a chain rectangle, the length of dims should 2 instead of {len(dims)}"
+            )
+
+        if lattice_shape == "cubic" and len(dims) != 3:
+            raise ValueError(
+                f"For a cubic lattice, the length of dims should 3 instead of {len(dims)}"
+            )
+
+        lattice_obj = Lattice(lattice_shape, nx.grid_graph(dims))
+        return lattice_obj
+
+    if lattice_shape == "triangle":
+        if len(dims) != 2:
+            raise ValueError(
+                f"For a triangle lattice, the length of dims should 2 instead of {len(dims)}"
+            )
+        lattice_obj = Lattice(lattice_shape, nx.triangular_lattice_graph(dims[0], dims[1]))
+        return lattice_obj
+
+    if lattice_shape == "honeycomb":
+        if len(dims) != 2:
+            raise ValueError(
```

Comment:
Can we simplify this? It will be highly changing to maintain, and likely can be simplified to remove the multiple nested branching statements.

Reply:
Done!